src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionArguments.cs

```diff
@@ -4,10 +4,13 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using Microsoft.Shared.Diagnostics;
 
 #pragma warning disable SA1111 // Closing parenthesis should be on line of last parameter
 #pragma warning disable SA1112 // Closing parenthesis should be on line of opening parenthesis
 #pragma warning disable SA1114 // Parameter list should follow declaration
+#pragma warning disable S3358 // Extract this nested ternary operation into an independent statement.
+#pragma warning disable S1698 // Consider using 'Equals' if value comparison is intended.
```

Comment:
This is for the comparer equality check. I'd like to either move this suppression down around that line of code only. Would `.Equals(comparer)` actually be better though, @stephentoub?

Reply:
Which line is this referring to? (In general, I'm not a fan of many of these "S" rules; they're not from the .NET SDK but some other analyzer library that's being pulled into extensions, and there are _lots_ of false positives.)

Reply:
If this is about https://github.com/dotnet/extensions/pull/6325/files#diff-7089a7412a27a3cecae56c30dbc91b04277c83d6aa5db5302d3616eb94e6aab7R65, == is fine, but it could also be ReferenceEquals if that would make folks happier.

Reply:
Leaving as is for now.